# CIS 443-01 - Analytics Programming
## Program 1
__Due__: Monday September 23 (by 11:59 PM)<br />
__Worth__: 50 pts.<br />
__Purpose__: Use a function and repetition control structure to demonstrate compounding interest.

### Directions
In the code cell below, write Python code to accomplish the following using what you've learned through Chapter 4. 


#### Part 1

Your code should allow the user to enter three values: a principle balance, an interest rate, and a number of years. 

For simplicity's sake, we will compound our interest once per year. In order to do so, create a function interest_calculator with three inputs, corresponding to the user inputs. They should have the default values 10000, .09, and 10, respectively. 

Your formula to calculate the interest will be Balance(Year + 1) = Balance * (1 + interest rate)

So for example, starting with 10,000 and a 5% interest rate, the formula is 10,000 * (1 + .05) = 10,500. The next year would be 10,500 * (1 + .05) = 11,025 and so on. 

Each year you will output the year number, and the value of the balance that year. 


In [52]:
from decimal import Decimal #Import decimal for interest rate
import locale #Import locale for currency formatting

locale.setlocale( locale.LC_ALL, '' ) #Set locale to current region

def interest_calculator (balance = 10_000, rate = .09, years = 10): #Create function with default values
    currBal = balance #Set current balance to initial balance
    for year in range(1, years + 1): #Loop through from 1 to the year parameter plus one (range's second value is exclusive)
        intComp = currBal * (1 + rate) #Calculate compounded interest for given year
        print(f"Year {year} compounded value is: {locale.currency(intComp, grouping=True)}") #Print year, compound interest
        currBal = intComp #Set the current balance for the next loop around to the compound interest

#Ask for user input. Note: No validation done
prinBal = int(input("Enter the principle balance: "))
intRate = Decimal(input("Enter the interest rate: "))
numYears = int(input("Enter the number of years: "))

#interest_calculator() #Used for testing default values
interest_calculator(prinBal, intRate, numYears) #Execute the function from earlier with user input
    

Enter the principle balance:  14339
Enter the interest rate:  .11
Enter the number of years:  10


Year 1 compounded value is: $15,916.29
Year 2 compounded value is: $17,667.08
Year 3 compounded value is: $19,610.46
Year 4 compounded value is: $21,767.61
Year 5 compounded value is: $24,162.05
Year 6 compounded value is: $26,819.87
Year 7 compounded value is: $29,770.06
Year 8 compounded value is: $33,044.77
Year 9 compounded value is: $36,679.69
Year 10 compounded value is: $40,714.46


### Example Output
<div class="alert alert-block alert-info">
Enter the principle balance:  14339<br>
Enter the interest rate:  .11<br>
Enter the number of years:  10<br>
Year 1 compounded value is \$15916.29<br>
Year 2 compounded value is \$17667.08<br>
Year 3 compounded value is \$19610.46<br>
Year 4 compounded value is \$21767.61<br>
Year 5 compounded value is \$24162.05<br>
Year 6 compounded value is \$26819.87<br>
Year 7 compounded value is \$29770.06<br>
Year 8 compounded value is \$33044.77<br>
Year 9 compounded value is \$36679.69<br>
Year 10 compounded value is \$40714.46
</div>

#### Part 2
**Answer the following**:

How many years of this calculation do you have to use before you first get 38,000, at 5% interest, starting with 7,500?<br />
34<br />
100,000?<br />
54<br />
1,000,000?<br />
101<br />

Double click the cell in Jupyter to edit (__5 pts.__)



Be sure to save and exit your Jupyter Notebook and Shutdown Jupyter Lab (from the __File__ menu) before you submit your notebook on Blackboard for grading.